core/src/com/unciv/logic/city/PopulationManager.kt

```diff
@@ -109,34 +111,63 @@ class PopulationManager {
         addPopulation(-population + count)
     }
 
-    internal fun autoAssignPopulation(foodWeight: Float = 1f) {
+    internal fun autoAssignPopulation() {
+        cityInfo.cityStats.update()  // calculate current stats with current assignments
+        // save temporaries
+        cityInfo.currentSurplusFood = cityInfo.foodForNextTurn().toFloat()  // save temporary for calculations
+        cityInfo.currentGPPBonus = 0
+        for (unique in cityInfo.getMatchingUniques(UniqueType.GreatPersonPointPercentage))
+            if (cityInfo.matchesFilter(unique.params[1]))
+                cityInfo.currentGPPBonus += unique.params[0].toInt()
+        // Sweden UP
+        for (otherCiv in cityInfo.civInfo.getKnownCivs()) {
+            if (!cityInfo.civInfo.getDiplomacyManager(otherCiv).hasFlag(DiplomacyFlags.DeclarationOfFriendship))
+                continue
+
+            for (ourUnique in cityInfo.civInfo.getMatchingUniques(UniqueType.GreatPersonBoostWithFriendship))
+                cityInfo.currentGPPBonus += ourUnique.params[0].toInt()
+            for (theirUnique in otherCiv.getMatchingUniques(UniqueType.GreatPersonBoostWithFriendship))
+                cityInfo.currentGPPBonus += theirUnique.params[0].toInt()
+        }
```

Comment:
Isn't this straight up copy + pasted from `CityInfo:493::getGreatPersonPointsForNextTurn()`? I'd rather extract it to a new method and call that in both places, if we at any point in the future want to change this code we would otherwise have to do it in two places (and probably forget)

Reply:
Pulled out now